src/threechances.cpp

```diff
@@ -114,10 +110,38 @@ void ThreeChances::initialize(HWND hwnd) {
 	hud->setInitialPosition();
 	//monsterGrid->logLayout();
 	sword.setDirection(DOWN);
-
+	
 	return;
 }
 
+void ThreeChances::restartGame() {
+	// Clear mv
+
+	std::vector<Entity*> mv;
+	gameControl->setGameState(GAME_STATE::player);
+	gameControl->setMonsterVec(mv);
+
+	// initialize level grid
+	levelGrid = new LevelGrid;
+	levelGrid->initialize(1);
+
+	// initialize monster grid
+	monsterGrid = new MonsterGrid;
+
+	player.setHealth(PLAYER_DATA.health);
+	player.setMovesLeft(PLAYER_DATA.moves);
+	player.setX(playerNS::X);
+	player.setY(playerNS::Y);
```

Comment:
Will it work if you dont set the x and y for player? since it doesnt change anyway.